Street Fighter 2 Turbo implementation (#44)
Tourmi/Retro-ML: Games/Retro_ML.StreetFighter2Turbo/README.md

> # Super Mario Kart Plugin

Reviewer: Whole file is just copied over from Super Mario Bros. Update for Street Fighter

# Street Fighter 2 Turbo plugin
Plugin by Tourmi & Jaewongtongsoup
-------
Training AIs to beat Street Fighter 2 Turbo fights automatically.

## Configuration

### Plugin Configuration
* **Internal Clock Tick Length (Frames)**
  * The time (in frames) that it takes for the internal clock to move to its next state.
* **Internal Clock Length**
  * The total amount of nodes the internal clock has.
  * The total time it takes for the clock to do a full cycle is equal to `2^Clock Length * Tick Length` frames.
 * **Frame Skip**
  * Amount of frames to skip for every AI evaluation

### Objectives

* **Combat**
  * The amount of points to attribute to an AI for its combat prowess at the end of a round. 

>   * The amount of points depends on the time of the fight and the HP delta between the AI and the enemy.

Reviewer: Don't forget to describe what the TimerInfluence does

* **End Round**
  * Points to attribute depending on the outcome of a round. Also influenced by a different multiplier for every outcome (Win, Lose, Draw)
    * Specific multipliers for distance traveled in the respective directions. These are multiplied with the objective's multiplier.
* **Stopped Fighting**

>   * Stops the current fight if the AI has not hit the enemy for a specific amount of frames. Used to speed up the training since we want the ai to be on the offensive the most he can.

Reviewer: ```suggestion
  * Stops the current fight if the AI has not hit the enemy for a specific amount of frames. Used to speed up training by skipping AIs that don't fight.
```

  * Player X : Player X position on the stage.
  * Enemy X : Enemy X position on the stage.
  * X Delta : X position delta between the player and the enemy.
  * Y Delta : Y position delta between the player and the enemy.
  * Enemy Direction : Boolean set to 1 if the enemy is facing left. 0 if the enemy is facing right.
  * Internal clock : Timed bias value. Alternates between on and off every couple frames.
  * Player Health : Value between 1 and 0 representing the percentage of health remaining to the player.
  * Enemy Health : Value between 1 and 0 representing the percentage of health remaining to the enemy.
  * Time Left : Value between 1 and 0 representing the percentage of the clock remaining.
  * Internal clock : Timed bias value. See `Internal Clock Tick Length (Frames)` and `Internal Clock Length` above
  * Bias : Bias value. Always on.

* **Output Nodes**
  * The output nodes represent each button on an SNES controller.
  * `A` : Middle Kick
  * `B` : Low Kick
  * `X` : Middle Punch
  * `Y` : Low Punch
  * `Left Shoulder` : High Punch
  * `Right Shoulder` : High Kick

>   * `Left` : Move Left / Block
>   * `Right` : Move Right

Reviewer: ```suggestion
  * `Left` : Move Left, blocks if moving away from enemy
  * `Right` : Move Right, blocks if moving away from enemy
```
